src/en/reference-hook-tools.md

```diff
@@ -0,0 +1,564 @@
+
+```no-highlight
+open 8000-8080/tcp
+```
+
+`open-port` will not have any effect if the service is not exposed, and may have
+a somewhat delayed effect even if it is. This operation is transactional, so 
+changes will certainly not be made unless the hook exits successfully.
+
+Juju also tracks ports opened across the machine and will not allow conflicts - 
+if another charm has already opened the port 
+(**or one or more ports in a range**) you have specified,
+your request will be ignored. 
+ 
+This command accepts and ignores `--format` for
+compatibility purposes, but it doesn't produce any output.
+
+### opened-ports
+
+The opened-ports hook tool lists all the ports currently opened 
+**by the running charm**. It does not, at the moment, include ports which may
+be opened by other charms co-hosted on the same machine
+[lp#1427770](https://bugs.launchpad.net/juju-core/+bug/1427770). 
+
+The command returns a list of one port or range of ports per line, with the port
+number followed by the protocol (tcp or udp).
+
+For example, running `opened-ports` may return:
+
+```no-highlight
+70-80/tcp
+81/tcp
+```
+
+!!! Note: opening ports is transactional (i.e. will take place on successfully 
+exiting the current hook), and therefore `opened-ports` will not return any
+values for pending `open-port` operations run from within the same hook. 
+
+
+
+
+### relation-get
+
+`relation-get` reads the settings of the local unit, or of any remote unit, 
+in a given relation (set with `-r`, defaulting to the current relation 
+identifier, as in `relation-set`). The first argument specifies the settings 
+key, and the second the remote unit, which may be omitted if a default is 
+available (that is, when running a relation hook other than -broken).
+
+If the first argument is omitted, a dictionary of all current keys and values
+will be printed; all values are always plain strings without any 
+interpretation. If you need to specify a remote unit but want to see all 
+settings, use `-` for the first argument.
+
+The environment variable `JUJU_REMOTE_UNIT` stores the default remote unit:
+
+```bash
+echo $JUJU_REMOTE_UNIT
+ mongodb/2
+```
+
+Getting the settings of the default unit in the default relation is done with:
+
+```no-highlight
+relation-get
+ username: jim
+ password: "12345"
+```
+
+To get a specific setting from the default remote unit in the default relation
+you would instead use:
+
+    
+```no-highlight
+relation-get username
+ jim
+```
+
+To get all settings from a particular remote unit in a particular relation you
+specify them together with the command:
+
+```no-highlight
+relation-get -r database:7 - mongodb/5
+ username: bob
+ password: 2db673e81ffa264c
+```
+
+Note that `relation-get` produces results that are _consistent_ but not
+necessarily _accurate_, in that you will always see settings that:
+
+  - were accurate at some point in the reasonably recent past
+  - are always the same within a single hook run...
```

Comment:
i believe it was the original authors intent to indicate a link to the term below - ie are always the same *except* when. It doesn't quite work though. I may have a go at rewording that